pkg/nat/nat.go

```diff
@@ -29,8 +29,25 @@ type PortSet map[Port]struct{}
 // 80/tcp
 type Port string
 
-func NewPort(proto, port string) Port {
-	return Port(fmt.Sprintf("%s/%s", port, proto))
+func NewPort(proto, port string) (Port, error) {
+	// Check for parsing issues on "port" now so we can avoid having
+	// to check it later on.
+
+	portInt, err := ParsePort(port)
+	if err != nil {
+		return "", err
+	}
+
+	return Port(fmt.Sprintf("%d/%s", portInt, proto)), nil
+}
+
+// NewPortNoError is the same as NewPort but ignores any parsing error.
+// This should ONLY be used by testcases where we know we don't have any
+// parsing issues.  Just because its annoying to have to check (or ignore)
+// the error each time when we know its correct.
+func NewPortNoError(proto, port string) Port {
```

Comment:
I don't think we need this, and is a bit dangerous to to add functions just to support a test case.
Tests can just do `p, _` just as this function is, or add a test helper called `newPort` that does this.